common/src/main/java/org/apache/sedona/common/raster/RasterConstructors.java

```diff
@@ -45,6 +55,94 @@ public static GridCoverage2D fromGeoTiff(byte[] bytes) throws IOException {
         return geoTiffReader.read(null);
     }
 
+    /**
+    * Returns a raster that is converted from the geometry provided.
+     * @param geom The geometry to convert
+     * @param raster The reference raster
+     * @param pixelType The data type of pixel/cell of resultant raster
+     * @param value The value of the pixel of the resultant raster
+     * @param noDataValue The noDataValue of the resultant raster
+     *
+     * @return Rasterized Geometry
+     * @throws FactoryException
+    */
+    public static GridCoverage2D asRaster(Geometry geom, GridCoverage2D raster, String pixelType, double value, Double noDataValue) throws FactoryException {
+
+        DefaultFeatureCollection featureCollection = getFeatureCollection(geom, raster.getCoordinateReferenceSystem());
+
+
+        double[] metadata = RasterAccessors.metadata(raster);
+        // The current implementation doesn't support rasters with properties below
+        // It is not a problem as most rasters don't have these properties
+        // ScaleX < 0
+        if (metadata[4] < 0) {
+            throw new IllegalArgumentException(String.format("ScaleX %f of the raster is negative, it should be positive", metadata[4]));
+        }
+        // ScaleY > 0
+        if (metadata[5] > 0) {
```

Comment:
If the ScaleX < 0 and ScaleY > 0 then we would have to flip the raster. To keep the function simple, we decided to not do it for now. I consulted Kristin on this.

Reply:
Got it. But since your code does two separate if condition check, this logic only allows `ScaleX >=0 and ScaleY <= 0` and disabled all other possibility. Is this an expected behavior? @Kontinuation

Reply:
Yes. This is the most common case for north-up images. For any other cases, we need to flip the resulting raster.

Reply:
Got it. Thanks!